prody2/protocols/protocol_ensemble.py

```diff
@@ -62,43 +66,79 @@ def _defineParams(self, form):
-                      default=STRUCTURE,
+                      default=STRUCTURE, condition="inputType == %d" % STRUCTURE,
                       label="Reference structure selection type",
                       help='The reference structure can be a separate structure or indexed from the set')
 
         form.addParam('refStructure', PointerParam, label="Reference structure",
                       condition="refType == %d" % STRUCTURE,
-                      pointerClass='AtomStruct',
-                      help='Select an atomic model as the reference structure')
+                      pointerClass='AtomStruct', allowsNull=True,
+                      help='Select an atomic model as the reference structure. '
+                      'When using Dali, this is optional and is used for selecting atoms at the end.')
 
-        form.addParam('refIndex', IntParam, label="Reference structure index",
-                      condition="refType == %d" % INDEX,
-                      help='Select the index of the reference structure in the set, starting from 1')
+        form.addParam('refIndex', IntParam, label="Reference structure index", default=1,
+                      condition="refType == %d and inputType != %d" % (INDEX, INDEX),
+                      help='Select the index of the reference structure in the set, starting from 1. '
+                      'When using Dali, this is optional and is used for selecting atoms at the end.')
 
         form.addParam('seqid', FloatParam, default=0.,
                       expertLevel=LEVEL_ADVANCED,
-                      label="Sequence Identity Cut-off (%)",
+                      label="Sequence identity cutoff",
                       help='Alignment mapping with lower sequence identity will not be accepted.\n'
                            'This can be a number between 0 and 100 or a decimal between 0 and 1')
 
         form.addParam('overlap', FloatParam, default=0.,
                       expertLevel=LEVEL_ADVANCED,
-                      label="Overlap Cut-off (%)",
+                      label="Overlap cutoff",
                       help='Alignment mapping with lower sequence coverage will not be accepted.\n'
                            'This can be a number between 0 and 100 or a decimal between 0 and 1') 
 
-        form.addParam('matchFunc', EnumParam, choices=['bestMatch', 'sameChid'], 
-                      default=SAME_CHID,
+        form.addParam('matchFunc', EnumParam, choices=['bestMatch', 'sameChid', 'sameChainPos'], 
+                      default=SAME_CHID, condition="inputType == %d" % STRUCTURE,
                       expertLevel=LEVEL_ADVANCED,
                       label="Chain matching function",
                       help='See http://prody.csb.pitt.edu/manual/release/v1.11_series.html for more details.\n'
-                           'Custom match functions will be added soon.')    
+                           'Custom match functions will be added soon.')
```

Comment:
I usually add these as `#TODO` in the code.

Reply:
The user would wonder about them though because they exist in ProDy itself. I still need to figure out how to do it. Probably with chain selection wizards.